### DISCUSSION TITLE
How to disable this behavior on snowflake?

### DISCUSSION THREAD

**Discussion by mkague** [dbeaver/dbeaver#23181] [Q&A]
Hello, my DBeavert is connected to a Snowflake server and sends the same query for introspect numerous times. This impacts the credit usage of the server and an undesirable behavior. Please, how can I turn this off?

```
SELECT DISTINCT
  CASE TABLE_TYPE
    WHEN 'BASE TABLE' THEN 'TABLE' // For backward compatibility. 
    ELSE TABLE_TYPE
  END AS TABLE_TYPE
FROM INFORMATION_SCHEMA.TABLES;
```
Version 24.0.0.202403091004

JDBC 3.13.30

Java VM
OpenJDK 64-Bit Server VM

Java Version
17.0.6

Java Runtime
OpenJDK Runtime Environment

OCSP Mode
FAIL_OPEN

**Comment by LonwoLonwo** [comment]
Hello @mkague 

Could you please describe - in which moments this query is triggering? In the SQL Editor? In Data Editor? In Metadata editor?

**Reply by LonwoLonwo** [reply]
Also, it is a metadata query. Does it really impact the server's credit usage?

**Comment by mkague** [comment]
Hello! @LonwoLonwo.
 
This behavior seems to occur after selecting a text in the SQL Editor, for example, to copy and find/replace in different tabs.
It takes several minutes to run, and I have no doubt that it is consuming credits.
![image](https://github.com/dbeaver/dbeaver/assets/160027827/672be9b7-d00d-45db-bf1b-21b341c71c1d)

After the query, the command SHOW runs.
![image](https://github.com/dbeaver/dbeaver/assets/160027827/1e5f9ee1-5c27-45d5-8142-f75cc66d8dd3)

`show /* JDBC:DatabaseMetaData.getProcedures() */ procedures like 'INDEX_DELAY_TIME_BB_N2' in schema "<DATABASE>"."<SCHEMA>"
`
